components/eam/src/physics/rrtmgp/radiation.F90

```diff
@@ -871,6 +964,58 @@ subroutine radiation_init(state)
                      sampling_seq='rad_lwsw', flag_xyfill=.true.)
       endif
 
+      if (do_SPA_optics) then !initialize SPA
+         write(iulog,*) 'initializing SPA, iam = ', iam
+         !find current_month
+         call get_curr_date(year,month,day,tod)
+         current_month = month
+         write(iulog,*) 'current month is',current_month
+         if (month==12) then
+            next_month = 1
+         else
+            next_month = month + 1
+         end if
+
+         allocate(aerosol_optical_property(pcols,pver,nlwbands,begchunk:endchunk))
+         call get_aerosol_optical_property_from_file(current_month,'AER_TAU_LW','lwband',nlwbands,aerosol_optical_property)
+         call pbuf_set_field(pbuf, aer_tau_bnd_lw_mon_1_idx,aerosol_optical_property)
+         deallocate(aerosol_optical_property)
+         allocate(aerosol_optical_property(pcols,pver,nlwbands,begchunk:endchunk))
+         call get_aerosol_optical_property_from_file(next_month,'AER_TAU_LW','lwband',nlwbands,aerosol_optical_property)
+         call pbuf_set_field(pbuf,aer_tau_bnd_lw_mon_2_idx,aerosol_optical_property)
+         deallocate(aerosol_optical_property)
+
+         allocate(aerosol_optical_property(pcols,pver,nswbands,begchunk:endchunk))
+         call get_aerosol_optical_property_from_file(current_month,'AER_TAU_SW','swband',nswbands,aerosol_optical_property)
+         call pbuf_set_field(pbuf, aer_tau_bnd_sw_mon_1_idx,aerosol_optical_property)
+         deallocate(aerosol_optical_property)
```

Comment:
Of course if you decide to remove some of the deallocation steps you will need to make sure you keep the one that happens before you switch size from `lwbands` to `swbands`.

Alternatively, you can just declare `aerosol_optical_property_lw` and `aerosol_optical_property_sw` arrays at the top and not have to `allocate/deallocate` anything.

Reply:
Good idea, Aaron. I like the idea of just allocating both aerosol_optical_property_lw and aerosol_optical_property_sw at the top of the subroutine and just assigning new values for each property. I'm not sure whether zeroing out in between usage is needed - it is safer, but if `get_aerosol_optical_property_from_file` is guaranteed to produce a slab of data of the same size as the allocated chunk of memory (which I think is the case), there's no point and skipping the zeroing would be faster and cleaner.